frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -0,0 +1,692 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The signed phase implementation.
+
+use super::*;
+use codec::{Encode, HasCompact};
+use sp_arithmetic::traits::SaturatedConversion;
+use sp_npos_elections::{is_score_better, CompactSolution};
+use sp_runtime::Perbill;
+
+/// A raw, unchecked signed submission.
+///
+/// This is just a wrapper around [`RawSolution`] and some additional info.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, Default)]
+pub struct SignedSubmission<A, B: HasCompact, C> {
+	/// Who submitted this solution.
+	pub(crate) who: A,
+	/// The deposit reserved for storing this solution.
+	pub(crate) deposit: B,
+	/// The reward that should be given to this solution, if chosen the as the final one.
+	pub(crate) reward: B,
+	/// The raw solution itself.
+	pub(crate) solution: RawSolution<C>,
+}
```

Comment:
It's _much_ easier to read the code if generic parameters are meaningful.

```suggestion
pub struct SignedSubmission<AccountId, Balance: HasCompact, Currency> {
	/// Who submitted this solution.
	pub(crate) who: AccountId,
	/// The deposit reserved for storing this solution.
	pub(crate) deposit: Balance,
	/// The reward that should be given to this solution, if chosen the as the final one.
	pub(crate) reward: Balance,
	/// The raw solution itself.
	pub(crate) solution: RawSolution<Currency>,
}
```

Note: had to guess at what `B` and `C` should be.

Reply:
76c3387ce

Reply:
Sure